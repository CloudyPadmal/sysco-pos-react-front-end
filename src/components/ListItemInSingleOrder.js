import React, { Component } from 'react';

/**
 * Contains a single item view of a single order in my_orders list
 */
class ListItemInSingleOrder extends Component {

    render() {
        let { productTitle, description, quantity, price } = this.props.singleItem;
        return (
            <li className="list-group-item d-flex justify-content-right" style={{ margin: '0px', padding: '0px', width: '100%' }}>
                <div className="row" style={{ margin: '0px', width: '100%' }}>
                    <div className="col-10 card" style={{ borderRadius: '0px' }}>
                        <div className="card-body" style={{ padding: '5px 0px' }}>
                            <h6 className="card-title d-flex justify-content-right">{productTitle}</h6>
                            <p className="card-subtitle text-muted d-flex justify-content-right" style={{ textAlign: 'initial' }}>{description}</p>
                        </div>
                    </div>
                    <div className="col-2 card text-right" style={{ borderRadius: '0px' }}>
                        <p className="card-text" style={{ margin: '5px 0 0 0' }}>{quantity} x {price.toFixed(2)}</p>
                        <p className="card-link" style={{ margin: '0px' }}><b>Rs. {(quantity * price).toFixed(2)}</b></p>
                    </div>
                </div>
            </li>
        );
    }
}

export default ListItemInSingleOrder;