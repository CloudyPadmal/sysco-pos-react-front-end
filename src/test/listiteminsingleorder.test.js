import React from 'react';
import { mountWrap } from './enzyme';

import ListItemInSingleOrder from '../components/ListItemInSingleOrder';

describe('ListItemInSingleOrder', () => {

    it('displays item in a single order', () => {
        const wrapper = mountWrap(<ListItemInSingleOrder
            key={1244}
            singleItem={{ price: 34.44 }}
        />);
        expect(wrapper).toMatchSnapshot();
    });
});