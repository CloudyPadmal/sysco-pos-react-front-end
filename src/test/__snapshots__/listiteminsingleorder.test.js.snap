// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemInSingleOrder displays item in a single order 1`] = `
<ListItemInSingleOrder
  singleItem={
    Object {
      "price": 34.44,
    }
  }
>
  <li
    className="list-group-item d-flex justify-content-right"
    style={
      Object {
        "margin": "0px",
        "padding": "0px",
        "width": "100%",
      }
    }
  >
    <div
      className="row"
      style={
        Object {
          "margin": "0px",
          "width": "100%",
        }
      }
    >
      <div
        className="col-10 card"
        style={
          Object {
            "borderRadius": "0px",
          }
        }
      >
        <div
          className="card-body"
          style={
            Object {
              "padding": "5px 0px",
            }
          }
        >
          <h6
            className="card-title d-flex justify-content-right"
          />
          <p
            className="card-subtitle text-muted d-flex justify-content-right"
            style={
              Object {
                "textAlign": "initial",
              }
            }
          />
        </div>
      </div>
      <div
        className="col-2 card text-right"
        style={
          Object {
            "borderRadius": "0px",
          }
        }
      >
        <p
          className="card-text"
          style={
            Object {
              "margin": "5px 0 0 0",
            }
          }
        >
           x 
          34.44
        </p>
        <p
          className="card-link"
          style={
            Object {
              "margin": "0px",
            }
          }
        >
          <b>
            Rs. 
            NaN
          </b>
        </p>
      </div>
    </div>
  </li>
</ListItemInSingleOrder>
`;
